src/components/catalog-promo/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import TitleArrowLink from '@components/title-arrow-link';
+import clsx from 'clsx';
+import { useProfile } from '@hooks/use-profile';
+import api from '@services/api';
+import type { Category } from '@services/generated-api/data-contracts';
+import styles from './catalog-promo.module.scss';
+
+const CatalogPromo = () => {
+	const [categories, setCategories] = useState<Category[]>([]);
+	const [numberToShow, setNumberToShow] = useState(6);
+	const { isMobileScreen } = useProfile();
+
+	const findNumbersToShow = useCallback(() => {
+		if (isMobileScreen) {
+			setNumberToShow(6);
```

Comment:
Можно return вместе с сеттером написать:
```suggestion
			return setNumberToShow(6);
```